embedded_postgres.go

```diff
@@ -106,7 +106,14 @@ func (ep *EmbeddedPostgres) Start() error {
 			}
 		}
 
-		if err := decompressTarXz(defaultTarReader, cacheLocation, ep.config.binariesPath); err != nil {
+		tempBinariesPath := ep.config.binariesPath + ".temp"
+		// Discard whatever we had previously
+		os.RemoveAll(tempBinariesPath)
```

Comment:
additionally add a defer to remove the temp location so we clean up if decompressTarXz or os.Rename fail.